### DISCUSSION TITLE
Why laika convert `[test]` in to `<p><code>[test]</code><p>`?

### DISCUSSION THREAD

**Discussion by btomala** [planet42/Laika#389] [Q&A] [answered]
Why laika convert `[test]` in to `<p><code>[test]</code><p>`? I was expecting `<p>[test]</p>`. I don't know about that format, I test it in two other converters and it doesn't work like that. 

Is there any specific reason for this? 
Is there a simple way to ski this transformation?

**Comment by jenshalm** [comment]
Hello and welcome. I think you'd need to provide more context or a full example of what you are doing. This does not happen in normal use cases and I cannot reproduce it either. And assuming that this is supposed to be Markdown, this would be link syntax and it is not clear how you use it and what it is supposed to link to.

**Comment by btomala** [comment]
Hi, of course, sorry for not being specific. I try to transform short statements which are user inputs in Md to HTML.
I'm doing this: 

```
Transformer.from(Markdown).to(HTML).build.transform("[test]")
```

and I'm getting an error: 

```
Left(laika.parse.markup.DocumentParser$ParserError: Error parsing document '/': One or more error nodes in result:
  [1]: unresolved link id reference: test

  [test]
  ^) was not an instance of cats.data.Validated$Valid, but an instance of scala.util.Left
```
What I want for errors is just ignore them and take the raw text in that case so I'm adding custom config:
```
val config = OperationConfig(failOnMessages = MessageFilter.None)
val html   = Transformer.from(Markdown).to(HTML).withConfig(config).build.transform("[test]")
```
and the result of the transformation is
```
Right("<p><code>[test]</code></p>")
```

I'm not very familiar with this tool yet. So maybe I'm doing something wrong.

**Comment by jenshalm** [comment, marked as answer]
Ah, now I understand, thanks for adding some context. What you are doing is not strictly wrong, but quite unusual and therefore not one of the use cases supported out of the box. The `failOnMessages` setting you are using is mostly intended to be used for "Visual Debugging", meaning instead of aborting the transformation with an error, it will render the errors in place instead, e.g. when using the preview server of the sbt plugin. More details on how this is meant to be used can be found here: https://planet42.github.io/Laika/latest/02-running-laika/03-configuration.html#visual-debugging

Now what you want is neither fail on errors nor render them as errors, but treat them as basic text input, which is quite unusual. This is still possible, but you would need to dig a bit deeper in Laika's extension APIs, in particular "Overriding Renderers": https://planet42.github.io/Laika/latest/04-customizing-laika/06-overriding-renderers.html. You would need to match on `InvalidSpan` and render it as plain text instead.

**Reply by btomala** [reply]
Ok, thanks I try to override invalid spans then.
